toynet_manager.py

```diff
@@ -0,0 +1,218 @@
+                                                   remove=True, detach=True, volumes=vol)
+        self.running_containers[new_container.name] = new_container
+
+        return new_container.name
+
+    def build_mininet_container(self, dev=True, docker_file='dev.Dockerfile'):
+        '''
+        Builds the Toynet container from the local Dockerfile or dev.Dockerfile.
+
+        dev: bool - True indicates to build the development image, False
+                    indicates to build the production image
+        dockerfile: String - indicates the relative path to the Dockerfile to
+                             use for the image build
+
+        returns: True if successful build, False otherwise
+        '''
+
+        if dev:
+            try:
+                self.dev_image = self.client.images.build(path='.',
+                                                          dockerfile=docker_file,
+                                                          rm=True, tag='toynet-dev')
+            except docker.errors.APIError:
+                self.dev_image = None
+                return False
+        else:
+            try:
+                self.prod_image = self.client.images.build(path='.', dockerfile=docker_file,
+                                                           rm=True, tag='toynet')
+            except docker.errors.APIError:
+                self.prod_image = None
+                return False
+
+        return True
+
+    def kill_container(self, container_name):
+        '''
+        Kills the specified Docker container if running.
+
+        container_name: String - the name of the container to kill
+
+        returns: True if the container was killed, False otherwise
+        '''
+
+        res = None
+
+        try:
+            self.running_containers[container_name].kill()
+            del self.running_containers[container_name]
+            res = True
+        except (KeyError, docker.errors.APIError):
+            res = False
+
+        return res
+
+    @staticmethod
+    def check_cpu_availability(short_thresh=80.0, med_thresh=75.0, long_thresh=70.0):
+        '''
+        Polls the CPU load and determines if the system can safely start
+        another Toynet container.
+
+        short_thresh: float - the threshold for above which we consider the one
+                              minute average CPU load to be too high
+        med_thresh: float - the threshold for above which we consider the five
+                              minute average CPU load to be too high
+        long_thresh: float - the threshold for above which we consider the
+                             fifteen minute average CPU load to be too high
+
+        returns: bool - True if CPU load is below thresholds, False otherwise
+        '''
+
+        # CPU percent usage representation for the past 1, 5, and 15 minutes
+        loads = [load / psutil.cpu_count() * 100 for load in psutil.getloadavg()]
```

Comment:
@takakonishimura I don't think the Windows functionality will matter because we're not running toynet on Windows (it's on the container which is built from an Ubuntu image). 

If we had plans to run this on a Windows system it would need to be ran a minimum 5 seconds prior to show accurate non-zero information- but that info would be limited to the last 5 seconds. That is to say the windows behavior will not give accurate information on the 1/5/15 minute intervals until those intervals have passed from the initial run of getloadavg(). 

All of that said, it seems safe to assume that the resources _should_ be available when this is initially run.

Reply:
Found in the `psutil` [code](https://github.com/giampaolo/psutil/blob/6e494bd4024c3d25769688b53e95942e529392d6/psutil/__init__.py#L1898) that there is a 1:1 link to `os.getloadavg`, added error handling accordingly.

I don't think we intend to run our backend Flask server on Windows, do we? If so I can easily add to the `__init__` function (and even force a 5 second sleep before returning if we think setup latency is acceptable to guarantee good statistics on the first call).